proposals/lip-0003.md

```diff
@@ -75,35 +78,32 @@ Given the study of the issue presented in the previous section, this LIP propose
 
 ## Specification
 
-The changes will only affect the `Delegates.prototype.generateDelegateList` function mentioned above. These changes consist of removing the nested `for` loops and generating a new list of “delegate hashes” to be re-calculated and ordered every round. In particular, the proposed fix implements a new algorithm that goes as follows:
+The changes affect the `shuffleDelegateListForRound` function mentioned above. These changes consist of removing the nested `for` loops and generating a new list of “delegate hashes” to be re-calculated and ordered every round. In particular, the proposed fix is as follows:
 
-1. The variable `seedSource` is generated in the same way as now:
+1. The variable `seedSource` is defined as follows:
 
 	```js
-	var seedSource = modules.rounds.calc(height).toString();
+	const seedSource = randomSeed1;
 	```
+	where `randomSeed1` is computed as specified in [LIP 0022](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0022.md#random-seeds-computation). 
 
-2. One hash is generated per delegate using `seedSource` and the delegate public key (or ID). For simplicity, we propose to use the hashing function already utilised in the existing code, although the final choice here is left to the reference implementation. Example given in pseudo-code:
+2. One hash per delegate is generated using `seedSource` and the delegate address. For simplicity, we propose to use SHA-256, which is the hashing function already utilised in the existing function. Following the current implementation, an example in pseudo-code will be:
 
 	```
-	forEach delegate
-		delegate.pubKeyHash = createHash(seedSource + delegate.publicKey)
+	forEach delegate in delegateList
+		delegate.roundHash = hash(seedSource | delegate.address)
 	```
+	where `|` indicates bytes concatenation.
 
-3. The delegates are reordered according to the hashes generated in step 2. This can be done according to any criteria (in an ascending order, for example) as long as the list is sorted in a deterministic and consistent way. Hence, the sorting algorithm must be [stable](https://en.wikipedia.org/wiki/Sorting_algorithm#Stability) to generate the same list of delegates on every node on the network, even in the unlikely event of a hash collision in the second point. As an example, the `sort()` method in JavaScript can be used for this purpose.
-
-This will not only fix the uniformity issue (caused by the two issues raised in the [Motivation](#motivation) section), but it will also reduce complexity, allowing simpler, more human-readable code in any implementation.
+3. The delegates are reordered in lexicographical order according to the hashes generated in step 2.
```

Comment:
This last point could specify "In the unlikely event that several delegates have the same hash, they are ordered lexicographically by delegate address."
This will never be necessary, but for me it feels a bit better if all possibilities are covered.

Reply:
This is a nice idea and uniquely defines the ordering. It should also be a simple change of defining the comparator function used for sorting as something like
` (delegate1.roundHash< delegate2.roundHash) ||  (delegate1.roundHash == delegate2.roundHash &&  (delegate1.address< delegate2.address))``.

Reply:
thanks for the good point, added!